src/tactic/sort.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2022 Arthur Paulino, Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Arthur Paulino, Damiano Testa
+-/
+import data.polynomial.degree.definitions
+
+/-!  # A tactic for sorting sums  -/
+
+namespace tactic
+
+/--  Takes an `expr` and returns a list of its summands. -/
+meta def get_summands : expr → list expr
+| `(%%a + %%b) := get_summands a ++ get_summands b
+| a            := [a]
+
+section with_cmp_fn
+
+/--  Given compare function `cmp_fn : expr → expr → bool` and an expression `e`, in an additive
```

Comment:
```suggestion
/--  Given compare function `cmp_fn : expr → expr → bool` and an expression `e`, in an
```
Todo: reformulate this so that the "additive commutative semigroup" is more as a motivation.  As is, it seems a requirement.

Possibly "We use this function for expressions in an additive commutative semigroup..."